OrbitGl/TracepointTrack.cpp

```diff
@@ -8,6 +8,43 @@
 
 TracepointTrack::TracepointTrack(TimeGraph* time_graph) : EventTrack(time_graph) {}
 
+void TracepointTrack::Draw(GlCanvas* canvas, PickingMode picking_mode) {
+  if (!were_tracepoints_hit_per_thread_) {
```

Comment:
Would "has_tracepoints" have the same meaning as "were_tracepoints_hit_per_thread_" ? I think it would be clearer. Also, we should probably add a "HasTracepoints()" method and use that instead of caching the value?

if(!has_tracepoints_) {
  has_tracepoints_ = !GOrbitApp->GetCaptureData().GetTracepointsOfThread(thread_id_).empty();
}

or even a single line (the lookup is already done once per draw for the threads that don't have tracepoints, we could do it on every draw for all):

has_tracepoints_ = !GOrbitApp->GetCaptureData().GetTracepointsOfThread(thread_id_).empty();

Reply:
Thank you so much for your suggestion. Will update the code.